routing/router.go

```diff
@@ -1353,7 +1352,15 @@ func (r *ChannelRouter) processUpdate(msg interface{},
 		// perform any of the expensive checks below, so we'll
 		// short-circuit our path straight to adding the edge to our
 		// graph.
-		if r.cfg.AssumeChannelValid {
+		// We can also skip fetching the channel point if we created the
+		// announcement for an unconfirmed channel.
+		ourKey := r.selfNode.PubKeyBytes[:]
```

Comment:
We could possibly pass things along as an additional attribute, this way we don't need to leak the details of the "fake" short chan IDs into the router itself.